src/rules/unused_ignored_function_params.erl

```diff
@@ -49,14 +49,32 @@ check_function(FunctionNode) ->
     Clauses = erl_syntax:function_clauses(FunctionNode),
     ComputedResults =
         lists:foldl(fun(Clause, Result) ->
-                       Patterns = erl_syntax:clause_patterns(Clause),
-                       ClausePatterns = [pattern_to_integer(Pattern) || Pattern <- Patterns],
-                       check_unused_args(Result, ClausePatterns)
+                       case is_clause_a_nif_stub(Clause) of
+                           true ->
+                               Result; %% Discard NIF stubs!
+                           false ->
+                               Patterns = erl_syntax:clause_patterns(Clause),
+                               ClausePatterns =
+                                   [pattern_to_integer(Pattern) || Pattern <- Patterns],
+                               check_unused_args(Result, ClausePatterns)
+                       end
```

Comment:
```suggestion
Patterns = erl_syntax:clause_patterns(Clause),
ClausePatterns = [pattern_to_integer(Pattern) || Pattern <- Patterns, not is_clause_a_nif_stub(Clause)],
check_unused_args(Result, ClausePatterns)
```

Reply:
I don't like this suggestion. I understand how it works (i.e. by forcing `ClausePatterns` to be `[]` when `is_clause_a_nif_stub(Clause) == false`, but…
1. it is computing that function for every `Pattern` in `Patterns` and we do know that it will always return the same.
2. it is somehow implying that there might be patterns that are not part of the NIF stub, when we know that we're going to skip them all… an intent that @diegomanuel's original code shows better.